File: 2529. Maximum Count of Positive Integer and Negative Integer.py
class Solution(object):
    def maximumCount(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l = 0
        r = len(nums) - 1

        while l <= r:
            m = (l + r) / 2;
            if nums[m] >= 0:
                r = m - 1
            else:
                l = m + 1

        neg = l

        while l < len(nums) and nums[l] == 0:
            l += 1

        pos = len(nums) - l

        return max(neg, pos)

File: 2529. Maximum Count of Positive Integer and Negative Integer.java
class Solution {
    public int maximumCount(int[] nums) {
        int l = 0, r = nums.length - 1;

        while(l <= r){
            int m = (l + r) / 2;
            if(nums[m] >= 0) r = m - 1;
            else l = m + 1;
        }

        int neg = l;
        while(l < nums.length && nums[l] == 0) ++l;

        int pos = nums.length - l;

        return neg > pos ? neg : pos;
    }
}